crypto/fuzzer/fuzzer.c

```diff
@@ -96,82 +119,103 @@ void fuzzer_reset_state(void) {
 }
 
 void crash(void) {
-  // intentionally exit the program, which is picked up as a crash by the fuzzer
-  // framework
+  // intentionally exit the program
+  // the fuzzer framework treats this as a crash
   exit(1);
 }
 
+// IDEA are there advantages to turning this into a macro?
+//
+// check the memory area for memory information leaks if MSAN is available,
+// crash if problems are detected
+void check_msan(void *pointer, size_t length) {
+#if defined(__has_feature)
+#if __has_feature(memory_sanitizer)
+  // check `address` for memory info leakage
+  __msan_check_mem_is_initialized(pointer, length);
+#else
+  (void)pointer;
+  (void)length;
+#endif
+#endif
+}
+
+// simplify the pointer check after a var_pointer = malloc()
+#define RETURN_IF_NULL(var_pointer) \
+  if (var_pointer == NULL) {        \
+    return 0;                       \
+  }
```

Comment:
This should return `-1` I think.

Reply:
Good point. `return -1` is a minor optimization on newer libFuzzer variants to signal the fuzz engine to discard this input. In the current "abort early because something is wrong" role of `RETURN_IF_NULL`, this is meaningful in theory. 

In practice this path is only used once `malloc()` returns `NULL` pointers. If the system is out of new memory to allocate, the fuzzer is in a broken state where we mainly aim to avoid not triggering deadly bugs and performance is not important. The primary edge case that is likely relevant here are `malloc(0)` calls where optimization is meaningful.

Reply:
Resolved in https://github.com/trezor/trezor-firmware/pull/2449/commits/66db14e0f96d2a21e39d3ebaab85bbe1365780f7.